### ISSUE TITLE
Some failure cases about attribute assignment

### ISSUE THREAD

**Issue by XiaominLi1997** [TonyLianLong/LLM-groundedDiffusion#9]
Hi, thanks for your nice work, I have tried the demo, and the method does have strong reasoning abilities, but there are some failure cases in attribute assignment.

Given the following response from ChatGPT:
Caption: A cartoon painting of a man in red standing next to another woman in blue
Objects: [('a man in red', [80, 150, 100, 200]), ('a woman in blue', [200, 150, 100, 200])]
Background prompt: A cartoon painting

I obtained:
1. seed=4354
![image (7)](https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/25433111/4a725030-6c19-454b-a06d-8d750a393e98)
2. seed=3628
![image (8)](https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/25433111/184ebb5f-a470-4ed8-89f8-11bbc728d7ed)

 What do you think of the problem might be?

Thanks

**Comment by TonyLianLong** [comment]
You might be using our Github code that has an older copy of our demo. We will update our Github soon.

Our demo is currently more updated than our Github, so you could clone the demo and run locally https://huggingface.co/spaces/longlian/llm-grounded-diffusion/tree/main

You could also try it online: https://huggingface.co/spaces/longlian/llm-grounded-diffusion




Tried two things with our demo:

1. I used your prompt but increased the frozen ratio.

<img width="1160" alt="image" src="https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/1451234/5a3da54c-b8eb-4df5-933d-f7d4a1a113df">

The quality can improve further (by tuning other hyperparams or changing seeds), but I feel that with cartoon style SD typically puts fewer details (e.g., very few details on the faces).

However, the attribute binding should be right most of the time, with Standard guidance (Faster modes give lower guidance).

2. I removed the cartoon style.

<img width="1156" alt="image" src="https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/1451234/a7c2dbec-fb9a-4cc6-aa60-eb2b3fd9896d">

Note the faces are a little weird, but that's a typical SD issue with small faces that you will also see in the baseline.

**Comment by XiaominLi1997** [comment]
Hi, 
To clarify, the results mentioned above were obtained from running locally. 

I also tried your new demo https://huggingface.co/spaces/longlian/llm-grounded-diffusion, and the results depend on seeds. When I use the same seed as you, I can get correct and reasonable results. But when I use a different seed, failure cases occur. Please see the following results.
![image](https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/25433111/ad1c136b-dcd5-4f03-ad3d-330106050f5e)
![1693105283462](https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/25433111/3cd532d7-9be6-4355-afa5-eeb419b60869)

Of course, I agree with

> However, the attribute binding should be right most of the time, with Standard guidance (Faster modes give lower guidance).

Thanks.

**Comment by TonyLianLong** [comment]
It seems like the seed 4354 indeed gives a wrong association. Seed 4353 gives something right (still not beautiful, at a similar level to the SD baseline).

The seed 4354 even gets the association wrong without the man, so I would attribute this to the insufficient text-to-image object association learned from training.

Seed 4354 (only the woman):
<img width="1577" alt="image" src="https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/1451234/49e4ec92-7d4f-49d9-9c58-eaa38b307032">

Seed 4353 (only the woman):
<img width="1553" alt="image" src="https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/1451234/30e11497-135f-41c9-a101-bc6576a4947e">

Seed 4353 (man and the woman):
<img width="1549" alt="image" src="https://github.com/TonyLianLong/LLM-groundedDiffusion/assets/1451234/8b643b07-ae32-4d8d-855e-1db61559ca98">


Longer explanation:

I think your observation can be interpreted this way: SD only sees image and paired caption during training, not instance annotations, so the associations from text to objects/attributes in the image are learned implicitly (without direct supervision).

Since SD is mostly trained on photos, with photo LMD already works in your case.

I believe cartoons are only a small fraction of the training data in the original SD (and GLIGEN, if you use LMD+), and the learned association is probably only weakly learned in the original SD that it still misses the association in some seeds.

LMD (stage 2), as a training-free method, can guide the generation to a specific layout. However, if SD doesn't recognize the text-to-object association (between "a woman in blue" and the cartoon woman in blue), then it is hard to convey the information to the model.

A solution is to train a LoRA or fine-tune the model to give the model more control. This is how people adapt and strengthen SD with a special style/domain. This is also an opportunity for future research to improve (making the association stronger and making it correct even more often).

**Comment by XiaominLi1997** [comment]
Thanks for your helpful explanation, I have no more questions now.
I will close this issue.
